### ISSUE TITLE
[Accessibility] Banner component causes 2D scroll on mobile

### ISSUE THREAD

**Issue by ghost** [Shopify/polaris#4478]
## Issue summary

The Banner component causes horizontal scroll on mobile screen sizes – anything less than 490 pixels in width:

<details><summary><em><b>Screenshot</b></em></summary>

![https://screenshot.click/16-07-vw0gx-qzg02.png](https://screenshot.click/16-07-vw0gx-qzg02.png)

</details>

Additional scroll is less than 5 pixels but 2D scroll does not meet [Shopify Mobile standards](https://development.shopify.io/engineering/developing_at_Shopify/accessibility/mobile#2D_scrolling).

## Expected behavior

Banners should be limited to the maximum width on all screen sizes.

## Actual behavior

Banners cause a slight horizontal scroll on small screens. The issue is consistent throughout the Shopify Admin (product pages, settings pages, etc.)

## Steps to reproduce the problem

Browse the Shopify Admin on a phone or using desktop in a window less than 490 pixels.

1. Navigate to the settings section.
2. Edit something that will cause an error banner. For example, save the general settings page without an address.
3. The error banner will cause a slight horizontal scroll.

## Specifications

- Are you using the React components? (Y/N): Y – Banner
- Polaris version number: v6
- Browser: Firefox
- Device: Samsung Galaxy 10 (360 pixels width)
- Operating System: Android
